### PR TITLE
Gh1129

### PR DESCRIPTION
**Pull request by maxwellchandler**
Added a comparrison script that generates and compares d objects between two branches

### CODE REVIEW

**Comment by jacksonsj** [comment]
@maxwellchandler - I see you've created several top-level directories to store files for testing scripts. Per https://r-pkgs.org/data.html#other-data & https://r-pkgs.org/data.html#data-extdata, could you move these all within either `tests` or `inst/extdata`?

**Comment by maxwellchandler** [comment]
@jacksonsj Done. Some of these tests will not work on master due to new mocks needing to be created as the url signature changed for some of the calls. They will all pass in this branch.

**Review by jacksonsj — APPROVED** [review]

**Comment by jacksonsj** [comment]
@jason-p-pickering 7 @sam-bao - Could you both provide your review on this PR as well?

**Comment by sam-bao** [comment]
@maxwellchandler can you explain a bit how this is supposed to work. I tried to run a coverage report and got an error, and if I run devtools::test I also get a number of errors:

```
covr::report()
Error: Failure in `/private/var/folders/n4/z5xp1_813kg94bzlvgtw212m0000gn/T/Rtmpb6xnj4/R_LIBS7d535dfaea0/datapackr/datapackr-tests/testthat.Rout.fail`
39m
 2.   └─datapackr::createKeychainInfo(...)

[ FAIL 5 | WARN 2 | SKIP 0 | PASS 42 ]
Error: Test failures
Execution halted
> devtools::test()
Loading datapackr
[1] "USING PACKGE VERSION: 4.2.4"
* The library is already synchronized with the lockfile.
Testing datapackr
✓ |  OK F W S | Context
✓ |   3       | Test GetDataValueSet [0.4 s]                                                                                                  
x |   0 1     | check analytics                                                                                                               
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Error (test-checkAnalytics.R:5:3): check analytics
Error: object 'secrets' not found
Backtrace:
 1. datimutils::loginToDATIM(secrets, d2_session_name = "d2_session") test-checkAnalytics.R:5:2
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
x |   0 1     | create_play_spectrum_output                                                                                                   
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Error (test-create-spectrum-output.R:5:3): create_play_spectrum_output
Error: object 'secrets' not found
Backtrace:
 1. datimutils::loginToDATIM(secrets, d2_session_name = "d2_session") test-create-spectrum-output.R:5:2
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
x |   0 2     | test-get-datapack-name                                                                                                        
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Error (test-get-datapack-name.R:6:3): Can read a Datapack Name and UIDs
Error: could not find function "test_sheet"
Backtrace:
 1. base::file.copy(...) test-get-datapack-name.R:6:2

Error (test-get-datapack-name.R:30:3): Can error on an invalid regional DataPack UID
Error: could not find function "test_sheet"
Backtrace:
 1. base::file.copy(...) test-get-datapack-name.R:30:2
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
x |   0 1     | get mech view                                                                                                                 
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Error (test-getmechview.R:5:3): get mech view
Error: object 'secrets' not found
Backtrace:
 1. datimutils::loginToDATIM(secrets, d2_session_name = "d2_session") test-getmechview.R:5:2
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
x |   1 1     | test-handshake                                                                                                                
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Error (test-handshake.R:5:3): Can handshake template
Error: could not find function "test_sheet"
Backtrace:
 1. datapackr::handshakeFile(...) test-handshake.R:5:2
 2. datapackr::canReadFile(path) /Users/sam/Documents/GitHub/datapackr/R/handshakeFile.R:51:2
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
x |   0 1     | pack a cop 20 OPU datapack                                                                                                    
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Error (test-packcop20opu.R:5:3): pack a cop 20 OPU datapack
Error: object 'secrets' not found
Backtrace:
 1. datimutils::loginToDATIM(secrets, d2_session_name = "d2_session") test-packcop20opu.R:5:2
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
x |   0 1     | pack a cop 21 data pack                                                                                                       
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Error (test-packcop21opu.R:5:3): pack a cop 21 data pack
Error: object 'secrets' not found
Backtrace:
 1. datimutils::loginToDATIM(secrets, d2_session_name = "d2_session") test-packcop21opu.R:5:2
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
x |   0 1     | pull code list                                                                                                                
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Error (test-pullcodelist.R:5:3): pull code list
Error: object 'secrets' not found
Backtrace:
 1. datimutils::loginToDATIM(secrets, d2_session_name = "d2_default_session") test-pullcodelist.R:5:2
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
✖ |   0 1     | pull sites                                                                                                                    
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Error (test-pullsites.R:6:3): pull sites
Error: object 'secrets' not found
Backtrace:
 1. datimutils::loginToDATIM(secrets, d2_session_name = "d2_session") test-pullsites.R:6:2
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Maximum number of failures exceeded; quitting early.
You can increase this number by setting `options(testthat.progress.max_fails)` 

══ Results ═══════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════
Duration: 0.8 s

[ FAIL 10 | WARN 0 | SKIP 0 | PASS 4 ]
```

File: R/getSiteList.R [hunk 1]
```diff
@@ -67,9 +62,9 @@ getSiteList <- function(org_unit_uids,
       dplyr::filter(country_name %in% countries$country_name)
 
   # Tag PSNUs ####
-    siteList %<>%
+    siteList <- siteList %>%
       dplyr::left_join(countries, by = c("country_uid" = "id")) %>%
-      dplyr::left_join(levels, by = c("country_name")) %>%
+      dplyr::left_join(levels[ , !(names(levels) %in% "country_uid")], by = c("country_name")) %>%
```
Comment (sam-bao) [inline R/getSiteList.R:67]: I think this file was deprecated. I see a number of conflicts on this PR, please clear them.

File: tests/custom_datapackr/4.2.2/datapackr/DESCRIPTION [hunk 2]
```diff
@@ -0,0 +1,61 @@
+Package: datapackr
```
Comment (sam-bao) [inline tests/custom_datapackr/4.2.2/datapackr/DESCRIPTION:1]: Master is now on version 4.2.4, what do all of these 4.2.2 files provide us?

**Review by sam-bao — COMMENTED** [review]
@maxwellchandler gh1129 was mostly about having a script/process that executes as much datapackr code as possible. As of this moment I am not able to get a coverage report with this PR.

**Comment by maxwellchandler** [comment]
@sam-bao You have to build from source using this version of datapackr and then you run the tests

**Comment by sam-bao** [comment]
@maxwellchandler I am still seeing a conflict for getSiteList. Based on this PR OI believe getSiteList was deprecated: https://github.com/pepfar-datim/datapackr/pull/255/files. Please double check that this was deprecated and resolve the conflict.

**Comment by maxwellchandler** [comment]
@sam-bao Yes sorry about that I must have merged wrong, either way its fixed now

**Comment by maxwellchandler** [comment]
@sam-bao Devtools::test now works. The errors are tests that are failing ONLY due to code now.

**Comment by maxwellchandler** [comment]
@sam-bao You should expect 9 failures.

**Review by sam-bao — CHANGES_REQUESTED** [review]
@maxwellchandler I am still reviewing this, but wanted to get you a few comments now.

**Pull request closed without merging**